ORBIT/phases/install/quayside_assembly_tow/moored.py

```diff
@@ -0,0 +1,397 @@
+
+            transfer_moored_substructures_from_storage(
+                g,
+                self.assembly_storage,
+                self.distance,
+                self.active_group,
+                towing,
+                towing_speed,
+                **kwargs,
+            )
+
+    def initialize_queue(self):
+        """
+        Initializes the queue, modeled as a ``SimPy.Resource`` that towing
+        groups join at site.
+        """
+
+        self.active_group = simpy.Resource(self.env, capacity=1)
+        self.active_group.vessel = None
+        self.active_group.activate = self.env.event()
+
+    def initialize_support_vessel(self, **kwargs):
+        """
+        Initializes Multi-Purpose Support Vessel to perform installation
+        processes at site.
+        """
+
+        specs = self.config["support_vessel"]
+        vessel = Vessel("Multi-Purpose Support Vessel", specs)
+
+        self.env.register(vessel)
+        vessel.initialize(mobilize=False)
+        self.support_vessel = vessel
+
+        stabilization = self.config["towing_vessel_groups"][
+            "stabilization_vessels"
+        ]
+
+        install_moored_substructures(
+            self.support_vessel,
+            self.active_group,
+            self.distance,
+            self.num_turbines,
+            stabilization,
+            **kwargs,
+        )
+
+    @property
+    def detailed_output(self):
+        """"""
+
+        return {
+            "operational_delays": {
+                **{
+                    k: self.operational_delay(str(k))
+                    for k in self.sub_assembly_lines
+                },
+                **{
+                    k: self.operational_delay(str(k))
+                    for k in self.turbine_assembly_lines
+                },
+                **{
+                    k: self.operational_delay(str(k))
+                    for k in self.installation_groups
+                },
+                self.support_vessel: self.operational_delay(
+                    str(self.support_vessel)
+                ),
+            }
+        }
+
+    def operational_delay(self, name):
+        """"""
+
+        actions = [a for a in self.env.actions if a["agent"] == name]
+        delay = sum(a["duration"] for a in actions if "Delay" in a["action"])
+
+        return delay
+
+
+@process
+def transfer_moored_substructures_from_storage(
+    group, feed, distance, queue, towing_vessels, towing_speed, **kwargs
+):
+    """
+    Process logic for the towing vessel group.
+
+    Parameters
+    ----------
+    group : Vessel
+        Towing group.
+    feed : simpy.Store
+        Completed assembly storage.
+    distance : int | float
+        Distance from port to site.
+    towing_vessels : int
+        Number of vessels to use for towing to site.
+    towing_speed : int | float
+        Configured towing speed (km/h)
+    """
+
+    towing_time = distance / towing_speed
+    transit_time = distance / 20
```

Comment:
is 20 a hardcoded vessel transit speed?  why is transit time different than towing time - return to port?

Reply:
Oh good catch - I meant to have this be the vessel speed.

From talking with Aaron, it seemed that the towing speed is significantly slower than what the vessels can travel at unburdened. I also got the impression the towing speed is more dependent on the substructure (draft, Cd) than the specific vessels which is why I added the input for 'towing_speed'.

Reply:
Resolved